plugins/modules/droplet_action_power_on.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, Mark Mercado <mmercado@digitalocean.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+---
+module: droplet_action_power_on
+
+short_description: Power on a Droplet
+
+version_added: 0.3.0
+
+description:
+  - Power on a Droplet.
+  - View the API documentation at U(https://docs.digitalocean.com/reference/api/api-reference/#tag/Droplet-Actions).
+
+author: Mark Mercado (@mamercad)
+
+requirements:
+  - pydo >= 0.1.3
+  - azure-core >= 1.26.1
+
+options:
+  droplet_id:
+    description:
+      - A unique identifier for a Droplet instance.
+      - If provided, C(name) and C(region) are ignored.
+    type: int
+    required: false
+  name:
+    description:
+      - The name of the Droplet to act on.
+      - If provided, must be unique and given with C(region).
+    type: str
+    required: false
+  region:
+    description:
+      - The name of the Droplet to act on.
+      - Required with C(name).
+    type: str
+    required: false
+
+extends_documentation_fragment:
+  - digitalocean.cloud.common.documentation
+"""
+
+
+EXAMPLES = r"""
+- name: Power on a Droplet
+  digitalocean.cloud.droplet_action_power_on:
+    token: "{{ token }}"
+    state: present
```

Comment:
What does `state` mean for these individual droplet actions? Perhaps for the power ones alone, we should just have `droplet_action_power` with present/absent state meaning power off or power on. Or custom state values, to account for power on/power off/shutdown.

Reply:
> What does `state` mean for these individual droplet actions? Perhaps for the power ones alone, we should just have `droplet_action_power` with present/absent state meaning power off or power on. Or custom state values, to account for power on/power off/shutdown.

Yes, this is a good suggestion; having a single `digitalocean.cloud.droplet_action.power` with `state` in `{power_on,power_off,shutdown}` is syntactically much more intuitive (and prevents one from having to code or think about what it would mean for `digitalocean.cloud.droplet_action_power_on` with `state=absent` - yes it makes sense that it might be "turn it off" but then why do we have `digitalocean.cloud.droplet_action_power_off`, etc). I'll work on this change.

Reply:
> > What does `state` mean for these individual droplet actions? Perhaps for the power ones alone, we should just have `droplet_action_power` with present/absent state meaning power off or power on. Or custom state values, to account for power on/power off/shutdown.
> 
> Yes, this is a good suggestion; having a single `digitalocean.cloud.droplet_action_power` with `state` in `{power_on,power_off,shutdown}` is syntactically much more intuitive (and prevents one from having to code or think about what it would mean for `digitalocean.cloud.droplet_action_power_on` with `state=absent` - yes it makes sense that it might be "turn it off" but then why do we have `digitalocean.cloud.droplet_action_power_off`, etc). I'll work on this change.

This implementation has been pushed 🚀